src/transformers/models/pix2struct/image_processing_pix2struct.py

```diff
@@ -0,0 +1,267 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image processor class for PIX2STRUCT."""
+import math
+from typing import Dict, Optional, Union
+
+import numpy as np
+
+from ...image_processing_utils import BaseImageProcessor, BatchFeature
+from ...image_transforms import convert_to_rgb, normalize
+from ...image_utils import ChannelDimension, ImageInput, is_batched, to_numpy_array, valid_images
+from ...utils import TensorType, is_torch_available, is_vision_available, logging
+from ...utils.import_utils import requires_backends
+
+
+if is_vision_available():
+    import PIL
+
+if is_torch_available():
+    import torch
+
+
+logger = logging.get_logger(__name__)
+
+
+class Pix2StructImageProcessor(BaseImageProcessor):
+    r"""
+    Constructs a PIX2STRUCT image processor.
+
+    Args:
+        do_resize (`bool`, *optional*, defaults to `True`):
```

Comment:
This argument doesn't seem to be used at init

Reply:
Fixed in 8d5573f